### PR TITLE
refactor: add integration config for pylons, refactor distributed tracing helper usage

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
From #2223, it was found that the pylons integration did not declare its own integration config and instead relied on storing distributed_tracing in the middleware object. To unify integrations to follow the same pattern, an integration config was added to pylons with the distributed_tracing setting set to True by default, and changing the attribute itself to have getter/setters as a property within the app class. Additionally, the distributed tracing header activation was refactored to utilize the `trace_utils` helper `activate_distributed_headers`.

### CODE REVIEW

File: releasenotes/notes/pylons-int-config-b500a59f78297b88.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    Added distributed tracing setting to integration config for pylons.
```
Comment (P403n1x87) [inline releasenotes/notes/pylons-int-config-b500a59f78297b88.yaml:4]: missing newline at EOF

Comment (Kyle-Verhoog) [inline releasenotes/notes/pylons-int-config-b500a59f78297b88.yaml:4]: Might be good to be more specific here that we're just adding an environment variable to enable distributed tracing

**Review by Kyle-Verhoog — COMMENTED** [review]
lgtm besides the release note 🙂

File: releasenotes/notes/pylons-int-config-b500a59f78297b88.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    Added an environment variable to enable distributed tracing for pylons.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/pylons-int-config-b500a59f78297b88.yaml:4]: lol sorry for not being more clear with the previous review 🙈  I was thinking that we'd explicitly state the environment variable: `PYLONS_DISTRIBUTED_TRACING` else the reader of the release notes would have to go digging into the code to find out what it was

File: ddtrace/contrib/pylons/patch.py [hunk 3]
```diff
@@ -11,6 +11,14 @@
 from .middleware import PylonsTraceMiddleware
 
 
+config._add(
+    "pylons",
+    dict(
+        distributed_tracing=asbool(get_env("pylons", "distributed_tracing", default=True)),
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/pylons/patch.py:17]: We should have a test case for the new environment variable

File: releasenotes/notes/pylons-int-config-b500a59f78297b88.yaml (hunk 2 above)
Comment (Kyle-Verhoog) [reply releasenotes/notes/pylons-int-config-b500a59f78297b88.yaml:4]: We should also add it to the integration docs in `pylons/__init__.py`

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

File: ddtrace/contrib/pylons/__init__.py [hunk 4]
```diff
@@ -22,6 +22,22 @@
 Global Configuration
 ~~~~~~~~~~~~~~~~~~~~
 
+.. py:data:: ddtrace.config.pylons['distributed_tracing']
+
+   Whether to parse distributed tracing headers from requests received by your pylons app.
+
+   Can also be enabled with the ``DD_BOTOCORE_DISTRIBUTED_TRACING`` environment variable.
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/pylons/__init__.py:29]:
```suggestion
   Can also be enabled with the ``DD_PYLONS_DISTRIBUTED_TRACING`` environment variable.
```

File: ddtrace/contrib/pylons/patch.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/pylons/patch.py:17]: Hmm doesn't look like the test added tests the environment variable

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

File: ddtrace/contrib/pylons/__init__.py (hunk 4 above)
Comment (Yun-Kim) [reply ddtrace/contrib/pylons/__init__.py:29]: 🤦

Comment (Yun-Kim) [reply ddtrace/contrib/pylons/__init__.py:29]: I also used `override_config` in the test case to set the setting to False, but I can just directly reference the `config` object directly. Is there any difference between the two?

**Review by majorgreys — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**